### ISSUE TITLE
misc-use-internal-linkage should not warn about consteval functions

### ISSUE THREAD

**Issue by chfast** [llvm/llvm-project#122096]
`consteval` functions never end up in the machine code. The `misc-use-internal-linkage` check should omit them.

```cpp
consteval int f(int x) { return x; }
```

https://godbolt.org/z/b37fohnEY

**Comment by chfast** [comment]
No problem, still waiting for the build :)
